apps/src/aiTutor/chatApi.ts

```diff
@@ -92,12 +94,18 @@ export async function getChatCompletionMessage(
     {role: Role.USER, content: formattedQuestion},
   ];
   let response;
+
+  // Some level instructions include data that looks like PII, optionally skip the PII filter for those levels.
+  // For example s/csa6-2023/lessons/6/levels/7/sublevel/4
+  const skipPIIFilter = levelInstructions?.includes("email");
```

Comment:
This feels hack-y to me. Here's what I don't like about this solution: 
- we'll need to update this rule any time we find another exception (it's essentially hardcoded)
- it's very specific to the curriculum content in CSA
- it won't work with translated level instructions 
- I'm a little nervous about students sending real PII in scenarios where we've skipped the PII filter 

On the other hand, I like that it's not "all or nothing" and we still mostly have PII filtering coverage. 

Very open to other suggestions for how we might handle false positives in a less brittle way!

Reply:
Hey Erin, my first thought is, why not blank out the email from the message before sending it to the PII filter?

my second thought is, wait a minute, why are we doing PII filtering on the instructions in the first place? shouldn't we just be running it on the student's entry? In your pilot writeup you mentioned students sometimes pasting the instructions into the chat prompt, but I'm not sure if that's what's going on here.

Reply:
to be more specific, I would suggest using a simple regex to identify email addresses rather than looking for the word "email", perhaps copying the exact one we use in our PII filter: https://github.com/code-dot-org/code-dot-org/blob/504257222dd02b9f9e11e09de50c18e7e6b62996/lib/cdo/regexp.rb#L6

I would also think that this could be done on the backend, so that the PII logic can be consolidated to ruby code.

Reply:
Turns out the false positives with the PII filter are more widespread than flagging emails for these specific levels. The more common and negatively impactful issue is that Geocoder is flagging "for loops" as addresses. I've updated the PR to reflect a new, broader approach and reached out to Travis to triple check that this solution is viable.